WordPress/src/main/java/org/wordpress/android/util/LocaleManager.java

```diff
@@ -0,0 +1,151 @@
+package org.wordpress.android.util;
+
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.content.res.Configuration;
+import android.content.res.Resources;
+import android.os.Build;
+import android.preference.PreferenceManager;
+
+import java.util.Locale;
+
+import javax.annotation.Nullable;
+
+/**
+ * Helper class for working with localized strings. Ensures updates to the users
+ * selected language is properly saved and resources appropriately updated for the
+ * android version.
+ */
+public abstract class LocaleManager {
```

Comment:
I forgot to mention this during my review. I am curious about why this class is marked as `abstract`, is there a benefit that I am missing about it? All `abstract` tells me is that we might have a different implementation, something like `WordPressLocaleManager` but we want to be able to pass the `LocaleManager` around instead which I don't think is the case here. I'd love to learn if there are other benefits I didn't know about.

Reply:
It prevents people from instantiating an instance of this class since it's a utility class. I can remove it if you think it's misleading...I think it's one of those things where some developers do it and some do not.

Reply:
Even if we create an instance, we won't be able to do anything with it, but that's a good point. Just for consistency, I think it might still be better not to make it `abstract` as other such classes don't follow that pattern.

P.S: I actually don't know which one I'd prefer if I was working solo in a project 🤔

Reply:
Ok, will do :)

Reply:
Done